src/WorkspaceUpdater.ts

```diff
@@ -0,0 +1,50 @@
+import * as vscode from "vscode";
+import { BINARY_UPDATE_WORKSPACE_INTERVAL } from "./globals/consts";
+import sendUpdateWorkspaceRequest from "./binary/requests/workspace";
+import { Logger } from "./utils/logger";
+import { tabNineProcess } from "./binary/requests/requests";
+
+const INITIAL_DELAY = 3000;
+
+export class WorkspaceUpdater implements vscode.Disposable {
+  private updateInterval: NodeJS.Timeout | null = null;
+
+  constructor() {
+    void tabNineProcess.onReady.then(() => {
+      setTimeout(() => {
+        updateWorkspace();
+        this.updateInterval = setInterval(
+          updateWorkspace,
+          BINARY_UPDATE_WORKSPACE_INTERVAL
+        );
+      }, INITIAL_DELAY);
+    });
+  }
+
+  dispose() {
+    if (this.updateInterval) {
+      clearInterval(this.updateInterval);
+    }
+  }
+}
+
+function updateWorkspace() {
+  const rootPaths = vscode.workspace.workspaceFolders
```

Comment:
pls  check that it  returns all the workspace and only the opened workspaces 

from the doc 
```
List of workspace folders (0-N) that are open in the editor. undefined when no workspace has been opened.

Reply:
yes, I checked